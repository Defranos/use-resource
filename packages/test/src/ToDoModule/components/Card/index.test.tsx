import React from "react";
import renderer from "react-test-renderer";

import { ITodo } from "ToDoModule";
import Card from ".";

const mockCard: ITodo = {
  id: 1,
  name: "My first todo task",
  isComplete: false,
};

describe("[TODO] => Card", () => {
  it("Should handle completed status", () => {
    const tree = renderer.create(<Card {...mockCard} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should handle uncompleted status", () => {
    const tree = renderer.create(<Card {...mockCard} isComplete />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
