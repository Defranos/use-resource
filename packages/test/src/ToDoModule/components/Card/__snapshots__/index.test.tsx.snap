// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[TODO] => Card Should handle completed status 1`] = `
<div
  className="makeStyles-wrapper-1 makeStyles-wrapper-3"
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "fontSize": "1.2em",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      My first todo task
    </div>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root makeStyles-icon-2"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root makeStyles-icon-2"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
      />
    </svg>
  </div>
</div>
`;

exports[`[TODO] => Card Should handle uncompleted status 1`] = `
<div
  className="makeStyles-wrapper-1 makeStyles-wrapper-4"
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "fontSize": "1.2em",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      My first todo task
    </div>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root makeStyles-icon-2"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root makeStyles-icon-2"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
      />
    </svg>
  </div>
</div>
`;
